e2e/helpers/tableColumnSchemaEditor.ts

```diff
@@ -166,13 +166,17 @@ const enterColumnSchema = async (
 export const addColumnsFromColumnSchemaConfig = async (
   page: Page,
   columnConfigs: ColumnSchemaConfig[],
-  columnStartingIndex: number = 0,
 ) => {
   await test.step('add columns from column schema config', async () => {
     const addColumnButton = page.getByRole('button', {
       name: 'Add Column',
     })
 
+    const columnStartingIndex =
+      await test.step('calculate starting index', async () => {
+        return await page.getByPlaceholder('Column name').count()
+      })
+
     for (
```

Comment:
Calculate starting index from which to add new columns, rather than requiring user to pass the starting index.